services/batch-service.js

```diff
@@ -50,25 +50,44 @@ class BatchService {
             .filter(aFile => (aFile?.fileName) && aFile?.fileName.trim().length > 0)
             .map(file => [file?.fileName, file]));
         const succeededFiles = [];
-        for (const aFile of aBatch.files) {
-            if (!uploadFiles.has(aFile.fileName)) {
-                continue;
-            }
-            const aUploadFile = uploadFiles.get(aFile.fileName);
-            aFile.updatedAt = getCurrentTime();
-            if (aUploadFile?.succeeded) {
-                aFile.status = FILE.UPLOAD_STATUSES.UPLOADED;
-                succeededFiles.push(aFile);
-                continue;
+        const skippedFiles = files.filter(f=>f.skipped === true);
+        const skippedCount = skippedFiles.length
+        const isAllSkipped = skippedCount === files.length;
+        
+        if (!isAllSkipped) {
+            let updatedFiles = [];
+            for (const aFile of aBatch.files) {
+                if (!uploadFiles.has(aFile.fileName)) {
+                    continue;
+                }
+                const aUploadFile = uploadFiles.get(aFile.fileName);
+                if( aUploadFile.skipped === true){
+                    continue;
+                }
+                aFile.updatedAt = getCurrentTime();
+                if (aUploadFile?.succeeded) {
+                    aFile.status = FILE.UPLOAD_STATUSES.UPLOADED;
+                    succeededFiles.push(aFile);
+                }
+                else {
+                    aFile.status = FILE.UPLOAD_STATUSES.FAILED;
+                    aFile.errors = aUploadFile?.errors || [];
+                }
+                updatedFiles.push(aFile) 
             }
-            aFile.status = FILE.UPLOAD_STATUSES.FAILED;
-            aFile.errors = aUploadFile?.errors || [];
+            aBatch.files = updatedFiles;
+            aBatch.fileCount = updatedFiles.length;
+        }
+        else {
+            aBatch.files = [];
+            aBatch.fileCount = 0;
         }
+        
         // Count how many batch files updated from FE match the uploaded files.
-        const isAllUploaded = files?.length > 0 && succeededFiles.length === files?.length;
-        aBatch.status = isAllUploaded ? (aBatch.type=== BATCH.TYPE.METADATA ? BATCH.STATUSES.UPLOADING : BATCH.STATUSES.UPLOADED) : BATCH.STATUSES.FAILED;
+        const isAllUploaded = files?.length > 0 && (succeededFiles.length + skippedCount  === files?.length);
+        aBatch.status = isAllUploaded ? (aBatch.type=== BATCH.TYPE.METADATA && !isAllSkipped? BATCH.STATUSES.UPLOADING : BATCH.STATUSES.UPLOADED) : BATCH.STATUSES.FAILED;
         aBatch.updatedAt = getCurrentTime();
```

Comment:
Please move this line into asyncUpdateBatch function, so that we'll never forget to set updatedAt property, thanks!

Reply:
ok